### DISCUSSION TITLE
Inject timestamp to give the model more contextual awareness

### DISCUSSION THREAD

**Discussion by LisaMacintosh** [open-webui/open-webui#3151] [Ideas]
I see that open-webui already displays the time/date of previous prompts/responses in chat but is not actually part of the context sent to the model.

It would be nice to see open-webui inject the time stamps into the context to help improve its replies so that the bot may greet the user properly according to the time of day instead of hallucinating a greeting. 

i.e. Good morning, how can I help? (time is actually 8 PM)

I feel this would be a good feature to implement especially when voice call/video chatting matures enough to the point where timing becomes more important in conversation.

**Comment by tjbck** [comment]
`Tools` would meet your needs here!
